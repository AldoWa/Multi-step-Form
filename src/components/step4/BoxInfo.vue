<template>
  <div class="flex flex-col bg-gray-200 bg-opacity-50 rounded-xl p-4">
    <span class="text-xs text-gray-600 font-normal mb-1">{{ title }}</span>
    <span class="text-xs text-gray-900 font-medium">{{ value }}</span>
  </div>
</template>

<script setup lang="ts">
interface BoxInfoProps {
  title: string;
  value: string;
}

defineProps<BoxInfoProps>()
</script>
