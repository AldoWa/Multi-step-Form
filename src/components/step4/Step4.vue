<template>
  <div class="grid grid-cols-3 gap-y-5 gap-x-6">
    <BoxInfo
      title="Full name"
      :value="data.personalInformation.name"
    ></BoxInfo>
    <BoxInfo
      title="Email Address"
      :value="data.personalInformation.email"
    ></BoxInfo>
    <BoxInfo
      title="Phone Number"
      :value="data.personalInformation.phone"
    ></BoxInfo>
    <BoxInfo
      title="Portfolio/GitHub Link"
      :value="data.personalInformation.githubLink"
    ></BoxInfo>
    <BoxInfo
      title="Skill Level"
      :value="data.skill"
    ></BoxInfo>
    <BoxInfo
      title="Challenge Preference"
      :value="formattedPreference"
    ></BoxInfo>
  </div>
</template>

<script setup lang="ts">
import BoxInfo from './BoxInfo.vue';

import useInject from '../../composables/inject';
import { computed } from 'vue';

const { data } = useInject()

const formattedPreference = computed(() => {
  const preferesFormatted = {
    html: 'HTML/CSS/JS',
    react: 'ReactJs',
    angular: 'AngularJs',
    vue: 'Vue.js'
  }

  return preferesFormatted[data.value.preference]
})
</script>
