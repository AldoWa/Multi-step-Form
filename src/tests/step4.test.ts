import { describe, expect, it } from "vitest";

import { mount } from "@vue/test-utils";

import BoxInfo from "../components/step4/BoxInfo.vue";
import Step4 from "../components/step4/Step4.vue";

const generateData = () => ({
  data: {
    personalInformation: {
      name: 'Teste',
      email: 'teste@teste.com',
      phone: '79999999',
      githubLink: 'github.com/link'
    },
    skill: 'Beginner',
    value: {
      preference: 'angular'
    },
    preference: 'angular'
  }
})

describe("Step 4", () => {
  describe('BoxInfo', () => {
    it('Should match with the snapshot', () => {
      const wrapper = mount(BoxInfo, {
        props: {
          title: 'Title',
          value: 'Value'
        }
      })

      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('Step 4', () => {
    it('Should render all box with data', async () => {
      const wrapper = mount(Step4, {
        global: {
          provide: {
            data: generateData()
          }
        }
      })

      const [fullName, emailAddress, phoneNumber, _, skilLevel, preference] = wrapper.findAllComponents(BoxInfo)
      
      expect(fullName.props('value')).toBe('Teste')
      expect(emailAddress.props('value')).toBe('teste@teste.com')
      expect(phoneNumber.props('value')).toBe('79999999')
      expect(skilLevel.props('value')).toBe('Beginner')
      expect(preference.props('value')).toBe('AngularJs')
    })

    it('Should match with the snapshot', () => {
      const wrapper = mount(Step4, {
        props: {
          title: 'Title',
          value: 'Value'
        },
        global: {
          provide: {
            data: generateData()
          }
        }
      })

      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})