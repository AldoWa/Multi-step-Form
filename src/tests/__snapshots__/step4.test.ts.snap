// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Step 4 > BoxInfo > Should match with the snapshot 1`] = `"<div class="flex flex-col bg-gray-200 bg-opacity-50 rounded-xl p-4"><span class="text-xs text-gray-600 font-normal mb-1">Title</span><span class="text-xs text-gray-900 font-medium">Value</span></div>"`;

exports[`Step 4 > Step 4 > Should match with the snapshot 1`] = `
"<div class="grid grid-cols-3 gap-y-5 gap-x-6" title="Title" value="Value">
  <div class="flex flex-col bg-gray-200 bg-opacity-50 rounded-xl p-4"><span class="text-xs text-gray-600 font-normal mb-1">Full name</span><span class="text-xs text-gray-900 font-medium">Teste</span></div>
  <div class="flex flex-col bg-gray-200 bg-opacity-50 rounded-xl p-4"><span class="text-xs text-gray-600 font-normal mb-1">Email Address</span><span class="text-xs text-gray-900 font-medium">teste@teste.com</span></div>
  <div class="flex flex-col bg-gray-200 bg-opacity-50 rounded-xl p-4"><span class="text-xs text-gray-600 font-normal mb-1">Phone Number</span><span class="text-xs text-gray-900 font-medium">79999999</span></div>
  <div class="flex flex-col bg-gray-200 bg-opacity-50 rounded-xl p-4"><span class="text-xs text-gray-600 font-normal mb-1">Portfolio/GitHub Link</span><span class="text-xs text-gray-900 font-medium">github.com/link</span></div>
  <div class="flex flex-col bg-gray-200 bg-opacity-50 rounded-xl p-4"><span class="text-xs text-gray-600 font-normal mb-1">Skill Level</span><span class="text-xs text-gray-900 font-medium">Beginner</span></div>
  <div class="flex flex-col bg-gray-200 bg-opacity-50 rounded-xl p-4"><span class="text-xs text-gray-600 font-normal mb-1">Challenge Preference</span><span class="text-xs text-gray-900 font-medium">AngularJs</span></div>
</div>"
`;
